torch/csrc/distributed/c10d/init.cpp

```diff
@@ -971,16 +971,17 @@ that adds a prefix to each key inserted to the store.
                   const c10::intrusive_ptr<::c10d::Store>&,
                   int,
                   int,
-                  ::c10d::ProcessGroupNCCL::Options>(),
+                  const c10::intrusive_ptr<
+                      ::c10d::ProcessGroupNCCL::Options>&>(),
```

Comment:
this is a reference of the ptr, so it means someone else need it keep it alive?

Reply:
yes, I think the caller should keep it alive. Specifically I think this option is only needed during the initialization, and we don't need to share the ownership in the `ProcessGroupNCCL` instantiation, as it only record (copy) the fields `opTimeOut` and `isHighPriorityStream`. So after construction, the `Options` is not needed anymore. 

We can also pass by pointer directly, but I didn't see if this is necessary.

Reply:
For extra safety, is it fine to remove the reference to the pointer and just incur a refcount bump here?

Reply:
yep that's fine as well, I will update the PR to make this a refcount bump to ensure safety.